### PR TITLE
[ENH] Tidy up data_gen

### PR DESCRIPTION
**Pull request by TonyBagnall**
improved coverage, standardise names, better testing

first go at simplifying data generation for test functions. Main thing here is removing the duplication for make_3d_test_data/make_example_3d_numpy. Gone for a standard for collections of make_example_*

which gives
    "make_example_2d_numpy",
    "make_example_3d_numpy",
    "make_example_unequal_length",
    "make_example_nested_dataframe",
    "make_example_long_table",
    "make_example_multi_index_dataframe",

there is more to do here, but this is (more than) enough for one PR!

Note I have removed a test assertion from this test in transformers. I dont know why its started to sporadically fail, but I do not think you can guarantee the test

np.testing.assert_array_equal(np.unique(y_pred), np.unique(y_test))

A classifier can predict all of one class, so this is not a valid test of the pipeline in my opinion

```python
def test_FeatureUnion_pipeline():
    """Test pipeline with FeatureUnion."""
    # pipeline with segmentation plus multiple feature extraction

    steps = [
        ("segment", RandomIntervalSegmenter(n_intervals=1, min_length=2)),
        (
            "transform",
            FeatureUnion([("mean", mean_transformer), ("std", std_transformer)]),
        ),
        ("clf", DecisionTreeClassifier()),
    ]
    clf = Pipeline(steps)
    clf.fit(X_train, y_train)
    y_pred = clf.predict(X_test)

    assert y_pred.shape[0] == y_test.shape[0]
    # Removing this test since its not always true, a classifier can predict a subset
    # of the classes, there is no requirement for this to be true
    # np.testing.assert_array_equal(np.unique(y_pred), np.unique(y_test))
```

### CODE REVIEW

**Review by MatthewMiddlehurst — COMMENTED** [review]
few small things

File: aeon/testing/utils/data_gen/_series.py [hunk 1]
```diff
@@ -61,3 +61,56 @@ def _make_index(n_timepoints, index_type=None):
...
+    index_type=None,
+    make_X=False,
+    n_columns=1,
+    random_state=None,
+):
+    """Return test data for forecasting tests.
+
+    Parameters
+    ----------
+    n_timepoints : int, optional
```
Comment (chrisholder) [inline aeon/testing/utils/data_gen/_series.py:78]: Default parameters need to be document on the first line (remove optional and change to default = 50. Do for all the other params

File: aeon/testing/utils/data_gen/_series.py [hunk 2]
```diff
@@ -61,3 +61,56 @@ def _make_index(n_timepoints, index_type=None):
...
+
+
+def make_forecasting_problem(
+    n_timepoints=50,
+    all_positive=True,
+    index_type=None,
+    make_X=False,
+    n_columns=1,
+    random_state=None,
+):
```
Comment (chrisholder) [inline aeon/testing/utils/data_gen/_series.py:73]: type annotation would be nice

File: aeon/testing/utils/data_gen/_series.py [hunk 3]
```diff
@@ -61,3 +61,56 @@ def _make_index(n_timepoints, index_type=None):
...
+    make_X : bool, optional
+        Should X data also be returned, by default False
+    n_columns : int, optional
+        Number of columns of y, by default 1
+    random_state : inst, str, float, optional
+        Set seed of random state, by default None
+
+    Returns
+    -------
+    ps.Series, pd.DataFrame
```
Comment (chrisholder) [inline aeon/testing/utils/data_gen/_series.py:93]: If the is a tuple then they should be on seperate lines i.e.

```
Returns
-----------
pd.Series
    description...
pd.DataFrame
    description...
```

File: aeon/testing/utils/data_gen/forecasting.py [hunk 4]
```diff
@@ -1,19 +1,12 @@
...
 
 __author__ = ["mloning"]
-__all__ = [
-    "_get_expected_index_for_update_predict",
-    "_generate_polynomial_series",
-    "make_forecasting_problem",
-    "_get_expected_index_for_update_predict",
-    "make_forecasting_problem",
-]
+__all__ = []
```
Comment (chrisholder) [inline aeon/testing/utils/data_gen/forecasting.py:4]: Remove if empty

**Review by chrisholder — DISMISSED** [review]

File: aeon/classification/convolution_based/tests/test_arsenal.py [hunk 5]
```diff
@@ -16,22 +16,23 @@
...
         time_limit_in_minutes=0.25,
         contract_max_n_estimators=3,
         num_kernels=20,
     )
     arsenal.fit(X_train, y_train)
-
+    assert len(arsenal.estimators_) > 1
+    X_train, y_train = make_example_3d_numpy()
+    arsenal.fit(X_train, y_train)
     assert len(arsenal.estimators_) > 1
```
[the remark this answers is not in the document]
Comment (TonyBagnall) [reply aeon/classification/convolution_based/tests/test_arsenal.py:30]: not sure where that came from!

File: aeon/classification/hybrid/tests/test_hc.py [hunk 6]
```diff
@@ -14,9 +14,7 @@ def test_hc1_defaults_and_verbosity():
     HIVECOTEV1._DEFAULT_N_SHAPELETS = 10
     HIVECOTEV1._DEFAULT_MAX_ENSEMBLE_SIZE = 5
     HIVECOTEV1._DEFAULT_N_PARA_SAMPLES = 10
-
-    X, y = make_3d_test_data(n_cases=20, n_timepoints=10, n_labels=2)
-
+    X, y = make_example_2d_numpy(n_cases=20, n_timepoints=10, n_labels=2)
```
[the remark this answers is not in the document]
Comment (TonyBagnall) [reply aeon/classification/hybrid/tests/test_hc.py:17]: some ropey refactoring I think,.

File: aeon/testing/utils/data_gen/_collection.py [hunk 7]
```diff
@@ -210,6 +212,128 @@ def make_unequal_length_test_data(
...
+    df["value"] = pd.Series(vals)
+    return df
+
+
+def make_example_multi_index_dataframe(n_instances=50, n_channels=3, n_timepoints=20):
+    """Generate example collection as multi-index DataFrame.
+
+    Parameters
+    ----------
+    n_instances : int
```
[the remark this answers is not in the document]
Comment (TonyBagnall) [reply aeon/testing/utils/data_gen/_collection.py:314]: agreed, but its a breaking change and there is enough going on already to refactor/deprecate

File: aeon/testing/utils/data_gen/_collection.py [hunk 8]
```diff
@@ -76,17 +72,19 @@ def make_3d_test_data(
...
+    if return_y:
+        return X, y
+    return X
 
 
-def make_2d_test_data(
+def make_example_2d_numpy(
     n_cases: int = 10,
     n_timepoints: int = 8,
+    return_y: bool = True,
```
[the remark this answers is not in the document]
Comment (TonyBagnall) [reply aeon/testing/utils/data_gen/_collection.py:85]: made it last for them all

File: aeon/datasets/_data_generators.py [hunk 9]
```diff
@@ -129,9 +129,9 @@ def make_example_2d_numpy(
 
     Examples
     --------
-    >>> from aeon.datasets import make_example_2d_numpy
-    >>> data, labels = make_example_2d_numpy(n_cases=2, n_timepoints=6,
-    ...                                      return_y=True, n_labels=2, random_state=0)
+    >>> from aeon.testing.utils.data_gen import make_example_2d_numpy
+    >>> data, labels = make_example_2d_numpy(n_cases=2, n_timepoints=6, n_labels=2,
+    ...    random_state=0)
```
[the remark this answers is not in the document]
Comment (TonyBagnall) [reply aeon/datasets/_data_generators.py]: tough crowd! :)

**Review by MatthewMiddlehurst — APPROVED** [review]

**Pull request merged**